packages/adapters/bitcoin/src/connectors/UnisatConnector.ts

```diff
@@ -0,0 +1,265 @@
+      purpose: AddressPurpose.Payment,
+      publicKey
+    }))
+
+    return accountList
+  }
+
+  public async signMessage(params: BitcoinConnector.SignMessageParams): Promise<string> {
+    const protocol = params.protocol === 'bip322' ? 'bip322-simple' : params.protocol
+
+    return this.wallet.signMessage(params.message, protocol)
+  }
+
+  public async sendTransfer(params: BitcoinConnector.SendTransferParams): Promise<string> {
+    const network = this.getActiveNetwork()
+
+    if (!network) {
+      throw new Error('No active network available')
+    }
+
+    const from = (await this.wallet.getAccounts())[0]
+
+    if (!from) {
+      throw new Error('No account available')
+    }
+
+    const result = await this.wallet.sendBitcoin(
+      params.recipient,
+      Number(UnitsUtil.parseSatoshis(params.amount, network))
+    )
+
+    return result
+  }
+
+  public async signPSBT(
+    params: BitcoinConnector.SignPSBTParams
+  ): Promise<BitcoinConnector.SignPSBTResponse> {
+    const psbtHex = Buffer.from(params.psbt, 'base64').toString('hex')
+
+    const signedPsbtHex = await this.wallet.signPsbt(psbtHex)
+
+    let txid: string | undefined = undefined
+    if (params.broadcast) {
+      txid = await this.wallet.pushPsbt(signedPsbtHex)
+    }
+
+    return {
+      psbt: Buffer.from(signedPsbtHex, 'hex').toString('base64'),
+      txid
+    }
+  }
+
+  public async switchNetwork(caipNetworkId: string): Promise<void> {
+    const network = this.getNetwork(caipNetworkId)
+
+    await this.wallet.switchChain(network)
+  }
+
+  public request<T>(_args: RequestArguments): Promise<T> {
+    return Promise.reject(new MethodNotSupportedError(this.id, 'request'))
+  }
+
+  private bindEvents(): void {
+    this.unbindEvents()
+
+    this.wallet.on('accountsChanged', account => {
+      if (typeof account === 'object' && account && 'address' in account) {
+        this.emit('accountsChanged', [account.address])
+      }
+    })
+  }
+
+  private unbindEvents(): void {
+    this.wallet.removeListener('accountsChanged', account => {
+      if (typeof account === 'object' && account && 'address' in account) {
+        this.emit('accountsChanged', [account.address])
+      }
+    })
+  }
+
+  public static getWallet(params: UnisatConnector.GetWalletParams): UnisatConnector | undefined {
+    if (!CoreHelperUtil.isClient()) {
+      return undefined
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any, init-declarations
+    let wallet: any
+
+    switch (params.id) {
+      case 'unisat':
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        wallet = (window as any)?.unisat
+        break
+      case 'bitget':
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        wallet = (window as any)?.bitkeep?.unisat
+        break
+      case 'binancew3w':
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        wallet = (window as any)?.binancew3w?.bitcoin
```

Comment:
how is this related to Unisat? should it be here or on a separate integration?

Reply:
Bitget and BinanceWeb3 Wallet use Unisat Wallet API, so I thought we could "kill 3 birds with 1 stone". Happy to remove tho. Testing BinanceWeb3 wallet is a bit more demanding and complex